src/fieldnorm/mod.rs

```diff
@@ -26,3 +26,97 @@ pub use self::serializer::FieldNormsSerializer;
 pub use self::writer::FieldNormsWriter;
 
 use self::code::{fieldnorm_to_id, id_to_fieldnorm};
+
+#[cfg(test)]
+mod tests {
+    use crate::common::CompositeFile;
+    use crate::fieldnorm::FieldNormReader;
+    use crate::fieldnorm::FieldNormsSerializer;
+    use crate::fieldnorm::FieldNormsWriter;
+    use crate::{
+        directory::{Directory, RAMDirectory, WritePtr},
+        schema::{STRING, TEXT},
+    };
+    use once_cell::sync::Lazy;
+    use std::{
+        panic::{catch_unwind, AssertUnwindSafe},
+        path::Path,
+    };
+
+    use crate::schema::{Field, Schema, STORED};
+
+    pub static SCHEMA: Lazy<Schema> = Lazy::new(|| {
+        let mut schema_builder = Schema::builder();
+        schema_builder.add_text_field("field", STORED);
+        schema_builder.add_text_field("txt_field", TEXT);
+        schema_builder.add_text_field("str_field", STRING);
+        schema_builder.build()
+    });
+
+    pub static FIELD: Lazy<Field> = Lazy::new(|| SCHEMA.get_field("field").unwrap());
+    pub static TXT_FIELD: Lazy<Field> = Lazy::new(|| SCHEMA.get_field("txt_field").unwrap());
+    pub static STR_FIELD: Lazy<Field> = Lazy::new(|| SCHEMA.get_field("str_field").unwrap());
+
+    #[ignore]
+    #[test]
+    pub fn test_fieldnorm_bug() -> crate::Result<()> {
```

Comment:
That's an interesting use of ignore, but let's not do that.

Let's keep the test but let's give it the right values.

Reply:
(and change its name of course)